service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsController.java

```diff
@@ -61,6 +64,28 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(
         return AboutToStartOrSubmitCallbackResponse.builder().data(data).build();
     }
 
+    @PostMapping("/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody CallbackRequest callbackrequest) {
+        CaseDetails caseDetails = callbackrequest.getCaseDetails();
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
+            .errors(validate(caseDetails))
+            .build();
+    }
+
+    private List<String> validate(CaseDetails caseDetails) {
```

Comment:
Let's be more specific with the name of this method - `validateDocumentUpload` perhaps?

Reply:
will do :)